docs/docsite/rst/contributor_guide.rst

```diff
@@ -0,0 +1,222 @@
+
+README and other markdown files
+-------------------------------
+
+Markdown files (those with the extension ``.md``) can be found in several directories within the repository. These files are primarily aimed at developers and those browsing the repository, to explain or give context to the other files nearby.
+
+The main exception to the above is the ``README.md`` in the repository root. This file is more important because it provides introductory information and links for anyone browsing the repository, both on GitHub and on the collection's `Ansible Galaxy page <https://galaxy.ansible.com/community/hashi_vault>`_.
+
+Markdown files can be previewed natively in GutHub, so they are easy to validate by reviewers, and there are no specific tests that need to run against them.
+
+Collection docsite
+------------------
+
+The collection docsite is what you are reading now. It is written in reStructuredText (RST) format and published on the :ref:`ansible_documentation` site. This is where we have long-form documentation that doesn't fit into the other two categories.
+
+If you are considering adding an entirely new document here it may be best to open a GitHub issue first to discuss the idea and how best to organize it.
+
+Refer to the :ref:`Ansible style guide <style_guide>` for all submissions to the collection docsite.
+
+RST files for the docsite are in the ``docs/docsite/rst/`` directory. Some submissions may also require edits to ``docs/docsite/extra-docs.yml``.
+
+Only the ``extra-docs.yml`` file is validated by the collection's CI, and there is not yet any automated preview for the docsite rendering, but this is an area we are hoping to improve on.
+
+Advanced
+^^^^^^^^
+
+Docsite pages can be generated locally through a workaround technique. **This is not a supported method** but it may be helpful to get more rapid feedback on docsite changes, if you're comfortable at a command line.
+
+The process is:
+
+#. Clone `ansible/ansible <https://github.com/ansible/ansible/>`_ or a fork of it.
+#. Copy the ``.rst`` files you want to preview into that repository's ``docs/docsite/`` directory.
+#. Install the requirements needed to build the docsite (from the repository root):
+
+    .. code-block:: shell-session
+
+      $ pip install -r requirements.txt
+      $ pip install -r docs/docsite/requirements.txt
+
+#. You may also need to remove write permission from group and other:
+
+    .. code-block:: shell-session
+
+      $ chmod -R go-w docs/docsite/rst
+
+#. Build the docs:
+
+    .. code-block:: shell-session
+
+      $ make coredocs
+
+The rendered HTML docs should be available in ``docs/docsite/_build/html/`` and can be opened in a browser.
+
+Running tests locally
+=====================
+
+If you're making anything more than very small or one-time changes, you'll want to run the tests locally to avoid having to push a commit for each thing, and waiting for the CI to run tests.
```

Comment:
Sneaky future tense here:
```suggestion
If you're making anything more than very small or one-time changes, run the tests locally to avoid having to push a commit for each thing, and waiting for the CI to run tests.
```